metagraph/src/graph/representation/base/sequence_graph.cpp

```diff
@@ -420,6 +421,71 @@ void DeBruijnGraph::call_unitigs(const CallPath &callback,
     ::mtg::graph::call_sequences(*this, callback, num_threads, true, min_tip_size, kmers_in_single_form);
 }
 
+std::shared_ptr<const bit_vector> DeBruijnGraph::get_last() const {
+    bit_vector_dyn last_bv(max_index() + 1);
+    call_nodes([&](node_index v) {
+        std::pair<char, node_index> last;
+        call_outgoing_kmers(v, [&](node_index u, char c) {
+            last = std::max(last, std::pair{c, u});
+        });
+        last_bv.set(last.second, true);
+    });
+    return std::make_shared<bit_vector_dyn>(std::move(last_bv));
+}
+
+void DeBruijnGraph::row_diff_traverse(size_t num_threads,
+                                      size_t max_length,
+                                      const bit_vector &rd_succ,
+                                      sdsl::bit_vector *terminal) const {
+    sdsl::bit_vector visited(max_index() + 1);
+    auto finalised = visited;
+    std::vector<size_t> distance(max_index() + 1);
+    assert(terminal->size() == visited.size());
+    assert(rd_succ.size() == visited.size());
+    auto set_terminal = [&](int v) {
+        distance[v] = 0;
+        (*terminal)[v] = true;
+    };
+    call_nodes([&](node_index v) {
```

Comment:
Once we know this works, can we add `num_threads` as a parameter to `call_nodes` so we can do this with multiple threads?

The `set_bit`, `fetch_bit`, etc. methods used in `BOSS::row_diff_traverse` can be used here for thread-safe atomic operations on `sdsl::bit_vector`.

Reply:
Ack, might be best to get back to it after some initial version is landed.